src/CloudFoundry.VisualStudio/Forms/EditDialog.xaml.cs

```diff
@@ -221,7 +223,19 @@ private void Publish_Click(object sender, RoutedEventArgs e)
             if (package != null)
             {
                 package.Save();
-                Push(package);
+
+                var componentModel = (IComponentModel)CloudFoundry_VisualStudioPackage.GetGlobalService(typeof(SComponentModel));
+
+                IVsPackageInstallerServices installerServices = componentModel.GetService<IVsPackageInstallerServices>();
+
+                if (installerServices.IsPackageInstalled(currentProj, CloudFoundry_VisualStudioPackage.packageId) == false)
+                {
+                    MessageBoxHelper.DisplayWarning("Cannot publish to Cloud Foundry!\nCloudfoundry Msbuild Tasks is not installed in your project, please install it and try again");
```

Comment:
Please use string.Format, Environment.NewLine
Remove the exclamation mark from Cannot publish to Cloud Foundry!
fix spelling of Cloudfoundry Msbuild Tasks to Cloud Foundry MSBuild Tasks